exonum-java-binding-core/src/main/java/com/exonum/binding/blockchain/Blockchain.java

```diff
@@ -74,35 +77,41 @@ public long getHeight() {
   }
 
   /**
-   * Returns a proof list of transaction hashes committed in the block at the given height or
-   * an empty list if the block at the given height doesn't exist.
+   * Returns a proof list of transaction hashes committed in the block at the given height.
    *
    * @param height block height starting from 0
    * @throws IllegalArgumentException if the height is negative
+   * @return a proof list index containing block hashes for the given height,
+   *         or {@code Optional.empty()} if the block at given height doesn't exist
    */
-  public ProofListIndexProxy<HashCode> getBlockTransactions(long height) {
+  public Optional<ProofListIndexProxy<HashCode>> getBlockTransactions(long height) {
     return schema.getBlockTransactions(height);
   }
 
   /**
-   * Returns a proof list of transaction hashes committed in the block with given id or an empty
-   * list if the block with given id doesn't exist.
+   * Returns a proof list of transaction hashes committed in the block with given id.
    *
-   * @param blockId id of the block
+   * @return proof list of transaction hashes committed in the block with given id,
+   *         or {@code Optional.empty()} if the block with given id doesn't exist
    */
-  public ProofListIndexProxy<HashCode> getBlockTransactions(HashCode blockId) {
+  public Optional<ProofListIndexProxy<HashCode>> getBlockTransactions(HashCode blockId) {
     MapIndex<HashCode, Block> blocks = schema.getBlocks();
     Block block = blocks.get(blockId);
-    return getBlockTransactions(block.getHeight());
+    return block == null ? Optional.empty() : getBlockTransactions(block.getHeight());
```

Comment:
Could you propose the correct way to do that with Optional? If I understand your snippet correctly, it would return Optional<Optional<ProofListIndexProxy<HashCode>>> in case if block was found and Optional.empty() otherwise (so we can't just add `get()` in the end of this statement).

Reply:
Sorry, I missed that `getBlockTransactions` returns `Optional`. Next one will be correct
```
    return Optional.ofNullable(block)
        .map(Block::getHeight)
        .flatMap(this::getBlockTransactions);

```